pkg/instrumentors/manager.go

```diff
@@ -30,26 +30,24 @@ import (
 	"go.opentelemetry.io/auto/pkg/process"
 )
 
-var (
-	// Error message returned when unable to find all instrumentation functions.
-	errNotAllFuncsFound = fmt.Errorf("not all functions found for instrumentation")
-)
+// Error message returned when unable to find all instrumentation functions.
+var errNotAllFuncsFound = fmt.Errorf("not all functions found for instrumentation")
 
 // Manager handles the management of [Instrumentor] instances.
 type Manager struct {
 	instrumentors  map[string]Instrumentor
 	done           chan bool
 	incomingEvents chan *events.Event
-	otelController *opentelemetry.Controller
 	allocator      *allocator.Allocator
+	otelController *opentelemetry.Controller
 }
 
 // NewManager returns a new [Manager].
 func NewManager(otelController *opentelemetry.Controller) (*Manager, error) {
 	m := &Manager{
 		instrumentors:  make(map[string]Instrumentor),
 		done:           make(chan bool, 1),
-		incomingEvents: make(chan *events.Event),
+		incomingEvents: make(chan *events.Event, 10),
```

Comment:
why is this buffered to 10?

Reply:
am surprised that we didn't had buffer before. Without buffer, the process would wait for GRPC request to complete.  This would impact performance of the process. I choose 10 at random, but ideally we should make this configurable.

Reply:
cool, yeah maybe some benchmarking tests would be good to have at some point

Reply:
Added https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/334